### PR TITLE
Issue 488 backward couple thermal equations

### LINKED ISSUE #488: Backward couple thermal model
Need to put in temperature dependence of parameter in electrochemical submodels

### PR DESCRIPTION
**Pull request by Scottmar93**
# Description

Added in temperature dependence of parameters into models. 

Fixes #488

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Key checklist:

- [x] No style issues: `$ flake8`
- [ ] All tests pass: `$ python run-tests.py --unit`
- [x] The documentation builds: `$ cd docs` and then `$ make clean; make html`

You can run all three at once, using `$ python run-tests.py --quick`.

## Further checks: 

- [x] Code is commented, particularly in hard-to-understand areas
- [x] Tests added that prove fix is effective or that feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Scottmar93** [comment]
@martinjrobins, there is currently an issue with `simplify`. Fails test: 
`File "/home/user/PycharmProjects/PyBaMM/tests/unit/test_expression_tree/test_simplify.py", line 231, in test_symbol_simplify` 
Issue is that: 
`symbol.simplify().id != symbol.id`.

File: input/parameters/lithium-ion/electrolyte_conductivity_Capiglia1999.py [hunk 1]
```diff
@@ -9,6 +12,23 @@ def electrolyte_conductivity_Capiglia1999(c_e):
     properties of non-aqueous electrolyte solutions for rechargeable lithium batteries.
     Journal of power sources 81 (1999): 859-862.
     .. [2] http://www.cchem.berkeley.edu/jsngrp/fortran.html
+    Parameters
+    ----------
+    c_e: :class: `pybamm.Symbol`
```
Comment (tinosulzer) [inline input/parameters/lithium-ion/electrolyte_conductivity_Capiglia1999.py]: I think technically pybamm symbols aren't fed directly into this function? This function itself is only called when evaluating a `Function` node. So really the c_e that gets fed in is a number or array

File: input/parameters/lithium-ion/graphite_mcmb2528_diffusivity_Dualfoil.py [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
+      Returns
+      -------
+      :`pybamm.Symbol`
+         Solid diffusivity
+   """
+
+    D_ref = 3.9 * 10 ** (-14)
+    arrhenius = np.exp(E_D_s / R_g * (1 / T_inf - 1 / T))
+
+    correct_shape = 0 * sto
```
Comment (tinosulzer) [inline input/parameters/lithium-ion/graphite_mcmb2528_diffusivity_Dualfoil.py]: do we definitely need this?

File: pybamm/models/submodels/particle/fickian/fickian_single_particle.py [hunk 3]
```diff
@@ -37,12 +37,27 @@ def get_fundamental_variables(self):
...
+            "X-average " + self.domain.lower() + " particle concentration"
+        ]
+        T_k_av = variables["Average " + self.domain.lower() + " electrode temperature"]
+
+        if self.domain == "Negative":
+            D_s = self.param.D_n(c_s_xav, T_k_av)
+        elif self.domain == "Positive":
+            D_s = self.param.D_p(c_s_xav, T_k_av)
+
+        N_s_xav = D_s * self._flux_law(c_s_xav)
```
Comment (rtimms) [inline pybamm/models/submodels/particle/fickian/fickian_single_particle.py]: minor point, but should D be in the flux law and then have `self._flux_law(c,D)` ?

**Pull request merged**